docarray/array/mixins/setitem.py

```diff
@@ -72,15 +72,61 @@ def __setitem__(
             else:
                 self._set_doc_by_id(index, value)
         elif isinstance(index, slice):
+            # TODO: handle this case da[1:3] = DocumentAray.empty(10)
+            # where length of value is longer than the slice
```

Comment:
I also think raising exception is easier. But I think since this operation is currently allowed in `DocumentArrayMemory`, correct me if I'm wrong but from the discussion we had since this is doable we should handle this case also in other storages? We can make handle this case a lower priority? @numb3r3  @hanxiao

Reply:
Sure, I will remove the TODO and if such corner case comes up we can always open another issue if needed.